packages/flutter/lib/src/material/snack_bar_theme.dart

```diff
@@ -31,6 +31,12 @@ enum SnackBarBehavior {
   ///
   /// See <https://material.io/design/components/snackbars.html> for more details.
   floating,
+
+  /// This behavior will cause [SnackBar] to be shown below [FloatingActionButton]
+  /// and above `Scaffold.bottomNavigationBar`. In case [FloatingActionButton] is
+  /// placed to the top of the [Scaffold] setting this behavior is equal to set
+  /// it to `SnackBarBehavior.floating`.
+  floatingBelowFab,
```

Comment:
I think it is ok to make it possible for users to configure snackbars to behave differently from the material design specification, but I am not sure doing it this way is best.
These enums represent the material spec, and `floatingBelowFab` doesn't really fit here. This can also be breaking for folks that have switch statements in their code for each possible value of this enum.